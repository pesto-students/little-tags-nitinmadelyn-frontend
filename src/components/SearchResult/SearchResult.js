import React from 'react';
import maincss from '../../assets/css/main.css';
import searchresultcss from './searchresult.css';
import { Link } from 'react-router-dom';
import { LanguageContext, languageObj } from '../../context/language-context';
import { HeartOutline, Heart } from 'react-ionicons';
import {
  IntlProvider,
  FormattedMessage,
  FromattedHTMLMessage,
} from 'react-intl';
import flatten from 'flat';
import { useHistory, useParams } from 'react-router-dom';
import hoodieimg from '../../assets/img/hoodie.png';
import Product from './Product/Product';
import config from '../../config/config';
import axios from 'axios';

const SearchResult = (props) => {
  const searchKeyword = useParams();

  const history = useHistory();
  const [products, setProducts] = React.useState([]);
  const [loading, setLoading] = React.useState(true);

  React.useEffect(() => {
    setLoading(true);
    async function getProducts() {
      const response = await axios
        .post(`${config.apiEndPoint}/product/search`, {
          filters: [
            {
              key: 'price',
              value: [0, 1000],
            },
            {
              key: 'name',
              value: '',
            },
            {
              key: 'categoryId',
              value: `${searchKeyword.categoryId}`,
            },
            {
              key: 'subcategoryId',
              value: `${
                searchKeyword.subCategoryId ? searchKeyword.subCategoryId : ''
              }`,
            },
          ],
          language: 'en',
        })
        .catch((error) => console.log(error));

      if (response) {
        setProducts(response.data.items);
        setLoading(false);
      }
    }
    getProducts();
  }, [searchKeyword.text]);

  const loadingMessage = () => {
    return (
      <>
        <div
          className='alert alert-info test-center'
          style={{ display: loading ? '' : 'none' }}
        >
          Loading...
        </div>
      </>
    );
  };

  const noProduct = () => {
    return (
      <>
        <div
          className='alert alert-danger test-center'
          style={{ display: products.length == 0 && !loading ? '' : 'none' }}
        >
          No product found!
        </div>
      </>
    );
  };

  // React.useEffect(
  //   () => {
  //     var request = require('request');
  //     var options = {
  //       'method': 'POST',
  //       'url': config.apiEndPoint+'/product/search',
  //       'headers': { 'Content-Type': 'application/json' },
  //       body: JSON.stringify({"language": props.language})
  //     };
  //     request(options, function (error, response) {
  //       if (error) throw new Error(error);
  //       const results = JSON.parse(response.body);
  //       if(results.status === 'success'){
  //         setProducts(Object.values(results.items))
  //       } else {
  //         alert("There's some error")
  //       }
  //     });
  //   },
  //   [props.language]
  // );

  return (
    <LanguageContext.Consumer>
      {(language) => (
        <IntlProvider
          locale={language.language}
          messages={flatten(languageObj[language.language])}
        >
          <main>
            <section className='main-container'>
              <div className='row'>
                <div className='col span-1-of-4'>
                  <div className='filter-bg'>
                    <div className='row'>
                      <h3>
                        <FormattedMessage id='category.filters' />
                      </h3>
                      <hr />
                    </div>

                    <div className='row'>
                      <div className='col span-2-of-2'>
                        <h4>
                          <FormattedMessage id='category.allFilters.brand' />
                        </h4>
                        <div className='brand-box'>
                          <label>
                            <input type='checkbox' />
                            <span></span> Puma
                          </label>
                          <label>
                            <input type='checkbox' />
                            <span></span> Levi's
                          </label>
                          <label>
                            <input type='checkbox' />
                            <span></span> Pepe Jeans
                          </label>
                          <label>
                            <input type='checkbox' />
                            <span></span> Adidas
                          </label>
                          <label>
                            <input type='checkbox' />
                            <span></span> Wrangler
                          </label>
                        </div>
                      </div>
                    </div>
                    <hr />

                    <div className='row'>
                      <div className='col span-2-of-2'>
                        <h4>
                          <FormattedMessage id='category.allFilters.price' />
                        </h4>
                        <input
                          type='text'
                          name='minPrice'
                          placeholder='Min'
                          className='price-filter'
                        />
                        <input
                          type='text'
                          name='maxPrice'
                          placeholder='Max'
                          className='price-filter'
                        />
                      </div>
                    </div>
                    <hr />
                    <div className='row'>
                      <div className='col span-2-of-2'>
                        <h4>
                          <FormattedMessage id='category.allFilters.size' />
                        </h4>
                        <select>
                          <option>S</option>
                          <option>M</option>
                          <option>L</option>
                          <option>XL</option>
                          <option>XXL</option>
                          <option>XXL</option>
                        </select>
                      </div>
                    </div>
                    <hr />

                    <div className='row'>
                      <div className='col span-2-of-2'>
                        <h4>
                          <FormattedMessage id='category.allFilters.color' />
                        </h4>
                        <div className='color-box'>
                          <span className='small-square red'></span>
                          <span className='small-square white'></span>
                          <span className='small-square blue'></span>
                          <span className='small-square green'></span>
                          <span className='small-square black'></span>
                          <span className='small-square yellow'></span>
                          <span className='small-square orange'></span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div className='col span-3-of-4'>
                  <div className='filter-bg'>
                    <div className='row result-title'>
                      {searchKeyword.text && (
                        <h3>
                          <FormattedMessage
                            id='category.result'
                            values={{ name: searchKeyword.text }}
                          />
                        </h3>
                      )}
                      {!searchKeyword.text && <h3>Results</h3>}
                      <hr />
                    </div>

                    <div className='item-container'>
                      {loadingMessage()}
                      {noProduct()}
                      {products.map((product) => (
                        <Product key={product.id} product={product} />
                      ))}
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </main>
        </IntlProvider>
      )}
    </LanguageContext.Consumer>
  );
};

export default SearchResult;
