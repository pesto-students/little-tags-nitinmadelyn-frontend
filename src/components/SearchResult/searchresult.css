.main-container {
  background-color: #222121;
  color: #fff;
  padding: 0 40px;
}

.filter-bg {
  background-color: #000;
  color: #fff;
  width: 100%;
  border-radius: 5px;
  text-align: center;
  padding: 0 0 20px 0;
}

.filter-bg hr{
  width: 90%;
}
.filter-bg h4{
  margin: 10px 0;
}



.filter-bg input{
  position: relative;
}

.price-filter {
  width: 15%;
  margin: 0 10px;
}

.result-title {
  display: block;
  width: 100%;
  float: left;
  text-align: left;
}

.result-title h3 {
  padding-left: 50px;
}

.item-container {
  display: inline-block;
  margin: 30px 75px;
}

.item {
  display: inline-block;
  float: left;
  border-radius: 5px;
}

.item h2 {
  display: block;
  font-weight: normal;
  float: left;
  clear: both;
  padding: 0 10px;
  margin: 0;
}

.item-image {
  display: block;
  float: left;
  padding: 10px;
}

.item-price {
  display: block;
  float: left;
  padding: 10px;
  font-size: 130%;
  clear: both;
}

.item-add-to-cart {
  display: block;
  clear: both;
  padding: 10px;
  width: 92%;
  margin: 2% 4%;
  text-transform: uppercase;
  border: none !important;
}

select {
  padding: 10px;
    border-radius: 5px;
    border: 1px solid #fff;
    background-color: #000;
    color: #fff;
}

.color-box {
  width: 100%;
  padding: 0 50px;
  margin: 0 auto;
  text-align: center;
}

.brand-box {
  width: 100%;
  display: grid;
  float: left;
}

.brand-box label {
  display: block;
  float: left;
  padding: 3px 50px;
  text-align: left;
}

.brand-box input {
  margin: 0 10px;
  display: block;
  float: left;
}

.small-square {
  height: 15px;
  width: 15px;
  display: block;
  float: left;
  border: 1px solid #fff;
  margin: 5px;
}

.red { background-color: red;}
.white { background-color: #fff;}
.black { background-color: #000;}
.green { background-color: green;}
.yellow { background-color: yellow;}
.blue { background-color: blue;}
.orange { background-color: orange;}

/* Desktops, large screens */
@media only screen and (max-width: 1200px) {
  
}

/* Small screens, laptops */
@media only screen and (max-width: 1024px) {
  
}
/* iPads, Tablets */
@media only screen and (max-width: 768px) {
  
}

/* Mobile devices */
@media only screen and (max-width: 480px) {
  .google-btn {
    float: none !important;
    margin: 5px auto !important;
    width: 100% !important;
  }
  .facebook-btn {
    float: none !important;
    margin: 5px auto !important;
    width: 100% !important;
  }
  .login-btn{
    margin: 5px auto !important;
    width: 100%;
  }
  .login-box input{
    padding: 4px;
  }
  .mobile span{
    top: 18px !important;
  }
  .signup-link{
    float: none !important;
    top: 10px;
    position: relative;
  }
  .margin-top-40{
    margin-top: 40px;
  }
}

@media only screen and (max-width: 375px) {
  
}

@media only screen and (max-width: 320px) {
  
}