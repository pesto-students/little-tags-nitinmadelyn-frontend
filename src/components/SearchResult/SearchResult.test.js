import React from "react";
import renderer from "react-test-renderer";
import { Router, Route, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";
import SearchResult from "./SearchResult";

it("renders correctly when search by text", () => {
  const language = 'en';
  const tree = renderer
    .create(
      <Router history={createMemoryHistory({ initialEntries: ["/"] })}>
        <Route path="/search/shirt" component={SearchResult}>
          <SearchResult language={language} />
        </Route>
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
