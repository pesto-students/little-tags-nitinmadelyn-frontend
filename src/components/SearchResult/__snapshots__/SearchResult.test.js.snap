// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when search by text 1`] = `null`;
